src/extension.ts

```diff
@@ -523,6 +523,50 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(vscode.commands.registerCommand(commands.CommandsConsts.globalImportSnippets,
         async _ => handleCommand(() => commands.importSnippets(snippetsProvider))
     ));
+
+    context.subscriptions.push(vscode.languages.registerDocumentDropEditProvider('*', {
```

Comment:
I refactored your code a bit to reduce amount of if blocks and to use Snippet mimetype instead of text/uri-list. Check comment in Provider file

```
    context.subscriptions.push(vscode.languages.registerDocumentDropEditProvider('*', {
        async provideDocumentDropEdits(
            _document: vscode.TextDocument,
            position: vscode.Position,
            dataTransfer: vscode.DataTransfer,
            _token: vscode.CancellationToken
        ): Promise<vscode.DocumentDropEdit | undefined> {
            const dataItem = await dataTransfer.get('application/vnd.code.tree.snippetsProvider')?.asString();
            if (!dataItem) {
                return;
            }
            try {
                const parsedSource = JSON.parse(dataItem) as readonly Snippet[];
                // only accept one snippet (not a folder)
                if (parsedSource?.length !== 1 || parsedSource[0].folder) {
                    return;
                }
                const draggedSnippet = parsedSource[0];
                // same as open snippet command
                if (draggedSnippet.resolveSyntax === undefined) {
                    draggedSnippet.resolveSyntax = true;
                }
                if (draggedSnippet.resolveSyntax) {
                    vscode.commands.executeCommand("editor.action.insertSnippet", { snippet: draggedSnippet.value });
                } else {
                    const editor = vscode.window.activeTextEditor;
                    if (!editor) {
                        return;
                    }
                    editor.edit(edit => { edit.insert(position, draggedSnippet.value ?? ''); });
                }
            } catch {
                // throws error when parsing `dataItem?.value`, just skip
            }
        }
    }));
```